src/gmcp.cpp

```diff
@@ -0,0 +1,404 @@
+  j["deck"]["masking"] = persona->decker->masking;
+  j["deck"]["evasion"] = persona->decker->evasion;
+  j["deck"]["sensor"] = persona->decker->sensor;
+  j["deck"]["io"] = persona->decker->io;
+
+  // Active programs
+  j["deck"]["programs"] = json::array();
+  for (struct obj_data *soft = persona->decker->software; soft; soft = soft->next_content) {
+    json jsoft;
+
+    jsoft["name"] = GET_OBJ_NAME(soft);
+    jsoft["rating"] = GET_PROGRAM_RATING(soft);
+    jsoft["type"] = GET_PROGRAM_TYPE(soft);
+
+    j["deck"]["programs"].push_back(jsoft);
+  }
+  
+  // Persona info
+  j["persona"]["name"] = persona->name;
+  j["persona"]["condition"] = persona->condition;
+
+  // Dump the json to a string and send it.
+  std::string payload = j.dump();
+  SendGMCP(ch->desc, "Matrix.Deck", payload.c_str());
+}
+
+void SendGMCPCharInfo( struct char_data * ch )
+{
+  if (!ch || !ch->desc || !ch->desc->pProtocol->bGMCP) return;
+  json j;
+  
+  j["name"] = GET_CHAR_NAME(ch);
+  j["height"] = ((float)GET_HEIGHT(ch) / 100);
+  j["weight"] = GET_WEIGHT(ch);
+  j["tke"] = GET_TKE(ch);
+  j["rep"] = GET_REP(ch);
+  j["noto"] = GET_NOT(ch);
+  j["metatype"] = pc_race_types_for_wholist[GET_RACE(ch)];
+  j["pronouns"] = genders[ch->player.pronouns];
+  j["syspoints"] = GET_SYSTEM_POINTS(ch);
+
+  // Dump the json to a string and send it.
+  std::string payload = j.dump();
+  SendGMCP(ch->desc, "Char.Info", payload.c_str());
+}
+
+json SerializeRoomExits( struct room_data *room )
+{
+  json j;
+  
+  for (int door = 0; door < NUM_OF_DIRS; door++) {
+    if (!room->dir_option[door]) continue;
+    if (!room->dir_option[door]->to_room) continue;
+    if (room->dir_option[door]->to_room == &world[0]) continue;
+    if (IS_SET(room->dir_option[door]->exit_info, EX_HIDDEN)) continue;
+    
+    json exit = json::object();
+    exit["direction"] = exitdirs[door];
+    exit["to"] = room->dir_option[door]->to_room_vnum;
+    if (IS_SET(room->dir_option[door]->exit_info, EX_LOCKED)) {
+      exit["state"] = "LOCKED";
+    } else if (IS_SET(room->dir_option[door]->exit_info, EX_CLOSED)) {
+      exit["state"] = "CLOSED";
+    } else {
+      exit["state"] = "OPEN";
+    }
+
+    j.push_back(exit);
+  }
+
+  return j;
+}
+
+void SendGMCPExitsInfo( struct char_data *ch ) 
+{
+  if (!ch || !ch->desc || !ch->desc->pProtocol->bGMCP) return;
+  if (!ch->in_room) return;
+
+  json j;
+  j["exits"] = SerializeRoomExits(ch->in_room);
+  j["room_vnum"] = GET_ROOM_VNUM(ch->in_room);
+  j["room_name"] = GET_ROOM_NAME(ch->in_room);
+
+  // Dump the json to a string and send it.
+  std::string payload = j.dump();
+  SendGMCP(ch->desc, "Room.Exits", payload.c_str());
+}
+
+void SendGMCPRoomInfo( struct char_data *ch, struct room_data *room ) 
+{
+  if (!ch || !ch->desc || !ch->desc->pProtocol->bGMCP) return;
+  json j;
+
+  j["vnum"] = GET_ROOM_VNUM(room);
+  j["name"] = GET_ROOM_NAME(room);
+
+  // Zone info
+  struct zone_data *z = &zone_table[room->zone];
```

Comment:
Zone info is not available to players; let's not send this along. The names are semi-random like "Warboss's third zone", etc.